### PR TITLE
Feature/Read File

### PR DESCRIPTION
**Pull request by KarlHarnois**
Using `make test` prints out a bunch of Quick/Nimble warnings the first time it is run in an environment. Something we'll have to address later.

### CODE REVIEW

**Comment by KarlHarnois** [comment]
@Palleas ready for review

File: Sources/GrimoireKit/File.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+struct File {
+  static func read(_ path: String) throws -> String {
+    let path = "file://" + path
+    guard let url = URL(string: path) else {
```
Comment (Palleas) [inline Sources/GrimoireKit/File.swift]: `URL(fileURLWithPath:)` ?

File: Sources/GrimoireKit/File.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  }
+}
+
+extension File {
+  enum Error: LocalizedError {
+    case invalidFilePath(String)
+
+    var errorDescription: String? {
+      switch self {
+      case .invalidFilePath(let path):
```
Comment (Palleas) [inline Sources/GrimoireKit/File.swift]: I'm team `case let .invalidFilePath(path):` but I don't mind if you prefer that way.

File: Tests/GrimoireKitTests/FileSpec.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import Quick
+import Nimble
+import Foundation
+@testable import GrimoireKit
+
+final class FileSpec: QuickSpec {
+  override func spec() {
+    describe("read") {
+      context("when file path is faulty") {
+        let path = "fo  oo / bba r / bazz"
```
Comment (Palleas) [inline Tests/GrimoireKitTests/FileSpec.swift]: Not a big fan of `foo/bar/baz` kind of strings 🤷‍♂️

File: Tests/GrimoireKitTests/FileSpec.swift [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        it("throws the correct error") {
+          expect { try File.read(path) }.to(throwError { error in
+            if case let File.Error.invalidFilePath(actual) = error {
+              expect(actual).to(equal("file://" + path))
```
Comment (Palleas) [inline Tests/GrimoireKitTests/FileSpec.swift]: `expect(actual) == "file://\(path)"` ?

Comment (Palleas) [reply Tests/GrimoireKitTests/FileSpec.swift]: Should we implement `Equatable` for `File.Error` ? Would make the whole test clearer.

File: Tests/GrimoireKitTests/Fixtures/FileSpecFixture.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+foo
```
Comment (Palleas) [inline Tests/GrimoireKitTests/Fixtures/FileSpecFixture.txt]: Same here 😁

**Review by Palleas — APPROVED** [review]

**Comment by KarlHarnois** [comment]
@Palleas ready for a second round of review, do you prefer the generic strings Im using for `FileSpecFixture.txt` or would you rather having some simili content in there?

**Review by Palleas — APPROVED** [review]

**Comment by Palleas** [comment]
👍

**Pull request merged**